libogc/sha.c

```diff
@@ -48,11 +49,29 @@ static s32 SHA_ExecuteCommand(const ShaCommand command, sha_context* context, co
 {
 	ioctlv* params = (ioctlv*)iosAlloc(__sha_hid, sizeof(ioctlv) * 3);
 	s32 ret = -1;
+	void* buffer = NULL;
 	
 	if(params == NULL)
 		return -4;
 	
-	for (u32 i = 0; i < data_size; i += SHA_MSGBLOCK_SIZE) {
+	//if data is not 64-byte aligned, and not in mem1, we allocate a buffer to contain the data
```

Comment:
the caller would want to know about this extra copying in a low-latency context, so would be good to have this in the documentation, meaning i suppose a comment over the declaration or definition of each affected API function

Reply:
im currently working on documentation (comments in the header) and maybe redoing this bit to use the stack instead of a memalign.